packages/main/src/managers/ConnectionManager.ts

```diff
@@ -73,6 +74,15 @@ export default class ConnectionManager {
     return this.conn
   }
 
+  public async closeConn() {
+    if (this.conn) {
+      this.getConn().then(conn => {
+        this.isClose = true
+        conn.end()
+      })
+    }
+  }
```

Comment:
This is good! But you used both the `async` keyword and `.then`. Typically you use one or the other, and I recommend preferring `async` / `await`. So you could change the body of this method to `const conn = await this.getConn(); conn.end()`. (I know that I used `.then` in `getConn`; that is because I wanted to add a callback to the promise, and also hold a reference to the resulting promise.)

You should also move `this.isClose = true` to the top so that it is set immediately and unconditionally.